Makefile

```diff
@@ -23,7 +23,7 @@ setup:
 	$(shell "${PROTOBUF_SETUP}")
 	$(shell "${OPENSSL_SETUP}")
 	which cargo || curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-	which maturin || pip install maturin
+	which maturin || pip install maturin[patchelf]
```

Comment:
should this optional dependency be in [`pyproject.toml`](https://github.com/allenai/dolma/blob/705d358cc5d2467c8ae0a703d3d27aded99d9c8f/pyproject.toml#L108-L114) too?

Reply:
I think that would be a good idea. I'll check how setting it up in venv operates when it is included in `pyproject.toml`

Reply:
Documentation regarding patchelf only mentions installing it with maturin itself with
`pip install maturin[patchelf]`
https://www.maturin.rs/installation
https://www.maturin.rs/distribution

This leads me to believe the way i did it is the way to go without further changing `pyproject.toml` itself
I'm looking into other projects using maturin with patchelf to see how other packages handle it.

Reply:
https://github.com/search?q=maturin%5Bpatchelf%5D&type=code
It seems different projects approach it differently.
dolma is packaged and built with make files so it is necessary to be in the `Makefile` but i think having it in `pyproject.toml` as well can help with documentation and potential build alternatives as well.

I'll try to find the best way to implement it and add a commit.

Reply:
Added commit [2141d07](https://github.com/allenai/dolma/pull/24/commits/2141d07cbab0bb5d187e6b83c1e86b32c9b23035)